### PR TITLE
Updates documentation for data modules

### PR DESCRIPTION
**Pull request by vulder**
resolves se-passau/VaRA#571

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5236199&bid=17163751)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5236199&bid=17163751)

File: docs/source/vara-ts-api/data.rst [hunk 1]
```diff
@@ -2,6 +2,7 @@ Data handling
 =============
 
 * :ref:`reports`
+* :ref:`Handling utilities for generated report files`
 * :ref:`data handling utilities`
 
 Reports
```
Comment (boehmseb) [inline docs/source/vara-ts-api/data.rst:8]: I cannot comment there, but:
- in line 12 it should say `an interfact`, not `a interface`
- in line 13, there is a comma missing after `In each filename`

File: varats/data/cache_helper.py [hunk 2]
```diff
@@ -18,7 +18,10 @@
 
 class GraphCacheType(Enum):
     """
-    Cached dataframes for graphs.
+    Identifiers for cached plot dataframes. These identifiers allow plot to
+    refere to a possibly cached dataframe.  Identifiers can be used in multiple
+    plots that use the same or a similar dataframe layout, e.g., are based on
+    similar data.
```
Comment (boehmseb) [inline varats/data/cache_helper.py]:
```suggestion
    Identifiers for cached plot dataframes. These identifiers allow a plot to
    refer to a possibly cached dataframe.  Identifiers can be used in multiple
    plots that use the same or a similar dataframe layout, e.g., are based on
    similar data.
```

File: varats/data/cache_helper.py [hunk 3]
```diff
@@ -28,6 +31,12 @@ class GraphCacheType(Enum):
 def __get_data_file_path(data_id: tp.Union[GraphCacheType, str],
                          project_name: str) -> Path:
     """
+    Compose the identifier und project into a file path that points to the
```
Comment (boehmseb) [inline varats/data/cache_helper.py]:
```suggestion
    Compose the identifier and project into a file path that points to the
```

File: varats/data/report.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 """
 The Report module implements basic report functionalities and provides a
-minimal interface to implement report: BaseReport.
+minimal interface `BaseReport` to implement own reports.
```
Comment (boehmseb) [inline varats/data/report.py]:
```suggestion
minimal interface ``BaseReport`` to implement own reports.
```

File: varats/data/reports/commit_report.py [hunk 5]
```diff
@@ -12,6 +12,9 @@
 
 
 class FunctionInfo():
+    """
+    Encapsulates the informations gathered for a single functions.
```
Comment (boehmseb) [inline varats/data/reports/commit_report.py]:
```suggestion
    Encapsulates the information gathered for a single functions.
```

File: varats/data/reports/commit_report.py [hunk 6]
```diff
@@ -81,6 +96,10 @@ def edge_to(self) -> str:
 
 
 class FunctionGraphEdges():
+    """
+    A graph like structure that represent the connections between
+    ``FunctionInfo``'s.
```
Comment (boehmseb) [inline varats/data/reports/commit_report.py]:
```suggestion
    A graph like structure that represents the connections between
    ``FunctionInfo``s.
```

File: varats/data/reports/commit_report.py [hunk 7]
```diff
@@ -174,6 +196,17 @@ def get_file_name(project_name: str,
         """
         Generates a filename for a commit report with 'yaml'
         as file extension.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/reports/commit_report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/reports/commit_report.py [hunk 8]
```diff
@@ -174,6 +196,17 @@ def get_file_name(project_name: str,
...
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/reports/commit_report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/reports/blame_report.py [hunk 9]
```diff
@@ -163,6 +166,17 @@ def get_file_name(project_name: str,
         """
         Generates a filename for a commit report with 'yaml'
         as file extension.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/reports/blame_report.py [hunk 10]
```diff
@@ -163,6 +166,17 @@ def get_file_name(project_name: str,
         """
         Generates a filename for a commit report with 'yaml'
         as file extension.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/report.py [hunk 11]
```diff
@@ -253,7 +351,19 @@ def get_file_name(report_shorthand: str,
...
-        Generates a filename for a commit report
+        Generates a filename for a report file out the different parts.
+
+        Args:
+            report_shorthand: unique shorthand of the report
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/report.py [hunk 12]
```diff
@@ -253,7 +351,19 @@ def get_file_name(report_shorthand: str,
...
                       file_ext: str = ".txt") -> str:
         """
-        Generates a filename for a commit report
+        Generates a filename for a report file out the different parts.
+
+        Args:
+            report_shorthand: unique shorthand of the report
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/report.py [hunk 13]
```diff
@@ -280,6 +390,19 @@ def get_supplementary_file_name(report_shorthand: str,
                                     file_ext: str = "") -> str:
         """
         Generates a filename for a supplementary report file.
+
+        Args:
+            report_shorthand: unique shorthand of the report
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/report.py [hunk 14]
```diff
@@ -317,7 +451,18 @@ def get_file_name(project_name: str,
...
         """
-        Generates a filename for a commit report
+        Generates a filename for a report file.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/report.py [hunk 15]
```diff
@@ -317,7 +451,18 @@ def get_file_name(project_name: str,
                       extension_type: FileStatusExtension,
                       file_ext: str = ".txt") -> str:
         """
-        Generates a filename for a commit report
+        Generates a filename for a report file.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/reports/commit_report.py [hunk 16]
```diff
@@ -185,6 +218,18 @@ def get_supplementary_file_name(project_name: str, binary_name: str,
                                     info_type: str, file_ext: str) -> str:
         """
         Generates a filename for a commit report supplementary file.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/reports/commit_report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/reports/empty_report.py [hunk 17]
```diff
@@ -22,6 +22,17 @@ def get_file_name(project_name: str,
                       file_ext: str = ".txt") -> str:
         """
         Generates a filename for a commit report without any file ending.
+
+        Args:
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/reports/empty_report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/reports/empty_report.py [hunk 18]
```diff
@@ -22,6 +22,17 @@ def get_file_name(project_name: str,
...
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/reports/empty_report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/reports/taint_report.py [hunk 19]
```diff
@@ -23,6 +23,18 @@ def get_file_name(project_name: str,
                       file_ext: str = "") -> str:
         """
         Generates a filename for a taint report.
+
+        Args:
+            report_shorthand: unique shorthand of the report
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/reports/taint_report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/reports/taint_report.py [hunk 20]
```diff
@@ -23,6 +23,18 @@ def get_file_name(project_name: str,
...
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/reports/taint_report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/reports/env_trace_report.py [hunk 21]
```diff
@@ -22,6 +22,18 @@ def get_file_name(project_name: str,
                       file_ext: str = ".json") -> str:
         """
         Generates a filename for a dataflow analysis with json file type.
+
+        Args:
+            report_shorthand: unique shorthand of the report
+            project_name: name of the project for which the report was generated
+            binary_name: name of the binary for which the report was generated
+            project_version: version of the analysed project, i.e., commit hash
```
Comment (boehmseb) [inline varats/data/reports/env_trace_report.py]:
```suggestion
            project_version: version of the analyzed project, i.e., commit hash
```

File: varats/data/reports/env_trace_report.py [hunk 22]
```diff
@@ -22,6 +22,18 @@ def get_file_name(project_name: str,
...
+            project_uuid: benchbuild uuid for the experiment run
+            extension_type: to specifed the status of the generated report
```
Comment (boehmseb) [inline varats/data/reports/env_trace_report.py]:
```suggestion
            extension_type: to specify the status of the generated report
```

File: varats/data/revisions.py [hunk 23]
```diff
@@ -1,5 +1,8 @@
 """
-Module for handling the state of project revisions in VaRA.
+Module for handling revision specific files.  When analyzing project, result
```
Comment (boehmseb) [inline varats/data/revisions.py]:
```suggestion
Module for handling revision specific files.  When analyzing a project, result
```

File: varats/data/revisions.py [hunk 24]
```diff
@@ -1,5 +1,8 @@
...
+files are generated for specific project revisions.  This module provides
+function to manage and access these revision specific files, e.g., get all
```
Comment (boehmseb) [inline varats/data/revisions.py]:
```suggestion
functionality to manage and access these revision specific files, e.g., to get all
```

File: varats/data/revisions.py [hunk 25]
```diff
@@ -91,14 +94,16 @@ def get_processed_revisions_files(
...
         project_name: target project
         result_file_type: the type of the result file
-        file_name_filter: optional filter to exclude certain files,
-                            returns true; if the file_name should not be
-                            checked
+        file_name_filter: optional filter to exclude certain files, returns
+                          true; if the file_name should not be checked
+
+    Returns:
+        a list of file paths to correctly processed revision files.
```
Comment (boehmseb) [inline varats/data/revisions.py]:
```suggestion
        a list of file paths to correctly processed revision files
```

File: varats/data/revisions.py [hunk 26]
```diff
@@ -244,15 +258,15 @@ def get_supplementary_result_files(
     revisions are returned.
 
     Args:
-        project_name (str): target project
-        result_file_type (MetaReport): the type of the result file
-        revision (str): The revision for which the result files should
+        project_name: target project
+        result_file_type: the type of the result file
+        revision: The revision for which the result files should
                         be returned.
```
Comment (boehmseb) [inline varats/data/revisions.py]:
```suggestion
        revision: the revision for which the result files should
                        be returned
```

File: varats/data/revisions.py [hunk 27]
```diff
@@ -244,15 +258,15 @@ def get_supplementary_result_files(
...
-        suppl_info_type (str): Only include result files of the specified type
+        suppl_info_type: Only include result files of the specified type
```
Comment (boehmseb) [inline varats/data/revisions.py]:
```suggestion
        suppl_info_type: only include result files of the specified type
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

**Comment by vulder** [comment]
fixed all remarked issues

**Review by boehmseb — APPROVED** [review]

**Pull request merged**